server/src/proxyManagement.mjs

```diff
@@ -21,97 +21,114 @@
 import { parse } from 'url';
 import WebSocket from 'ws';
 
-let ws = null
-let responseObject = {}
+const wsMap = new Map();
+const wsMsgMap = new Map();
 
-async function initialize(callback, returnWS) {
-  if(ws) {
-    return true
+async function initializeAndSendRequest(returnWs, command) {
+  let outgoingWs = wsMap.get(returnWs);
+  /* Checks to see if ws connection is in map.
+   * If connection exists and is active, it will be used.
+   * Else, a new connection will be created and mapped to returnWs.
+   */
+
+  // Set up listeners once and only once
+  if (!outgoingWs) {
+    outgoingWs = await setupOutgoingWs(returnWs);
+
+    outgoingWs.on('message', (data) => {
+      const buf = Buffer.from(data, 'utf8');
+      const response = JSON.parse(buf.toString());
+
+      if (response.id === undefined) {
+        // In case of event, send the event to caller directly.
+        returnWs.send(buf.toString());
+      } else {
+        wsMsgMap.set(returnWs, buf.toString());
+      }
+    });
   }
-  const url = buildWSUrl()
-  ws = new WebSocket(url)
+
+  outgoingWs.send(command);
+
+  return new Promise(async (res, rej) => {
+    try {
+      const response = await getResponseMessageFromProxy(returnWs);
+      // Clear for next message
+      wsMsgMap.set(returnWs, null);
+      res(response);
+    } catch (err) {
+      rej('Timeout occurred');
```

Comment:
Can we make this message a little more specific? Something like "Timeout waiting for WS repsonse"